### ISSUE TITLE
ALE Should Not Run When Entering an Empty File

### ISSUE THREAD

**Issue by amarakon** [dense-analysis/ale#4284]
I added these two lines in my Neovim configuration (Lua) to not run ALE automatically after exiting insert mode:

```lua
vim.g.ale_lint_on_text_changed = "never"
vim.g.ale_lint_on_insert_leave = 0
```

If I enter into an empty Haskell file, (`neovim file.hs`), I will get an error message even though I haven't written anything yet.
I think it should not show any warnings or errors if the file is empty.

**Comment by hsanson** [comment]
ALE lints on these situations (see `:h ale-lint`):

* When you modify a buffer.                - g:ale_lint_on_text_changed
* On leaving insert mode.                  - g:ale_lint_on_insert_leave
* When you open a new or modified buffer.  - g:ale_lint_on_enter
* When you save a buffer.                  - g:ale_lint_on_save
* When the filetype changes for a buffer.  - g:ale_lint_on_filetype_changed

Maybe setting g:ale_lint_on_enter to zero will make ALE behave as you expect.

**Comment by amarakon** [comment]
That does work, but it also causes behavior I don't want.
If I enter a file that isn't empty, it will not lint because I set `g:ale_lint_on_enter` to zero.
I want it to only lint when the file isn't empty.

- It should not lint when writing to an empty file.
- It should not lint when entering an empty file.
- It should lint when entering a non-empty file.
- It should lint when writing to a non-empty file.

**Comment by hsanson** [comment]
I see, that is not supported by ALE.

**Comment by w0rp** [comment]
We should add an option for it. It's going in the milestone.
